src/core/entity/dbus_query.cpp

```diff
@@ -777,13 +753,56 @@ void DBusQuery::registerObjectRemovingObserver(
             log<level::DEBUG>("Can't handle interfaceRemoved signal",
                               entry("ERROR=%s", e.what()),
                               entry("SIGNAL=InterfaceRemoved"),
-                              entry("DBUS_OBJ=%s", message.get_path()));
-            return;
+                              entry("DBUS_OBJ=%s", objectPathStr.c_str()));
+            return false;
         }
+
+        return true;
     };
 
-    addObserver(this->getConnect()->createWatcher(rules::interfacesRemoved(),
-                                                  std::move(handler)));
+    DBusQuery::instanceRemoveHandlers.push_back(std::move(handler));
+}
+
+template <typename TInterfacesDict, typename TCacheUpdatingDict>
+bool DBusQuery::processGlobalSignal(sdbusplus::message::message& message,
```

Comment:
The result looks unused.

Reply:
Let's leave it as is so far. @ikmsk10 will later fix if needed.